Framework/Source/Sample.h

```diff
@@ -70,146 +72,79 @@ namespace Falcor
         Flags flags = Flags::None;                                  ///< Sample flags
     };
 
-    /** Bootstrapper class for Falcor.
-        User should create a class which inherits from Sample, then call Sample::run() to start the sample.
-        The render loop will then call the user's overridden callback functions.
+    /** Bootstrapper class for Falcor
+        Call Sample::run() to start the sample.
+        The render loop will then call the user's Renderer object
     */
-    class Sample : public Window::ICallbacks
+    class Sample : public Window::ICallbacks, public SampleCallbacks
     {
     public:
-        Sample();
-        virtual ~Sample();
-        Sample(const Sample&) = delete;
-        Sample& operator=(const Sample&) = delete;
-
         /** Entry-point to Sample. User should call this to start processing.
             On Windows, command line args will be retrieved and parsed even if not passed through this function.
             On Linux, this function is the only way to feed the sample command line args.
 
             \param[in] config Requested sample configuration
+            \param[in] pRenderer The user's renderer
             \param[in] argc Optional. Number of command line arguments
             \param[in] argv Optional. Array of command line arguments
         */
-        virtual void run(const SampleConfig& config, uint32_t argc = 0, char** argv = nullptr);
+        static void run(const SampleConfig& config, Renderer::UniquePtr& pRenderer, uint32_t argc = 0, char** argv = nullptr);
```

Comment:
Make `pRenderer` a const ref

Reply:
Move `argc` and `argv` to `SampleConfig`

Reply:
I think a const UniquePtr ref can't be used to assign the renderer to the sample via move ctor. Trying to do so gives the error that copy ctor for unique ptr is deleted.  Understanding based off [this stack overflow](https://stackoverflow.com/questions/41947020/making-a-const-unique-ptr-then-trying-to-stdmove-from-it-gives-the-same-error), let me know if that's incorrect.

Reply:
That's right, you can't make it const ref because the Sample takes ownership which requires a change to the referenced unique-ptr

Reply:
That's right, you can't make it const ref because the Sample takes ownership which requires a change to the referenced unique-ptr